crates/relayer/src/chain/ckb4ibc/utils.rs

```diff
@@ -165,19 +170,43 @@ pub fn get_search_key(script: Script) -> SearchKey {
         filter: None,
         with_data: Some(true),
         group_by_transaction: None,
+        script_search_mode: None,
     }
 }
 
-pub fn get_connection_capacity() -> Capacity {
-    Capacity::bytes(CONNECTION_CELL_CAPACITY as usize).unwrap()
-}
-
-pub fn get_channel_capacity() -> Capacity {
-    Capacity::bytes(CHANNEL_CELL_CAPACITY as usize).unwrap()
-}
-
-pub fn get_packet_capacity() -> Capacity {
-    Capacity::bytes(PACKET_CELL_CAPACITY as usize).unwrap()
+pub fn get_search_key_with_sudt(
+    script: Script,
+    symbol: &str,
+    network: NetworkType,
+) -> Result<SearchKey, Error> {
+    let sudt_code_hash = match network {
+        NetworkType::Mainnet => SUDT_CODE_HASH_MAINNET,
+        NetworkType::Testnet => SUDT_CODE_HASH_TESTNET,
+        _ => {
+            return Err(Error::other_error(format!(
+                "unsupported network: {network}"
+            )))
+        }
+    };
+    let owner_lockhash =
+        H256::from_str(symbol).map_err(|_| Error::other_error("invalid sUDT symbol".to_owned()))?;
+    let sudt_script = Script::new_builder()
+        .code_hash(sudt_code_hash.pack())
+        .hash_type(ScriptHashType::Type.into())
+        .args(owner_lockhash.as_bytes().to_vec().pack())
+        .build();
+    let filter = SearchKeyFilter {
+        script: Some(sudt_script.into()),
+        ..Default::default()
+    };
+    Ok(SearchKey {
+        script: script.into(),
+        script_type: ScriptType::Lock,
+        filter: Some(filter),
+        with_data: Some(true),
+        group_by_transaction: None,
+        script_search_mode: None,
```

Comment:
Please use `IndexerScriptSearchMode::Exact` search mode https://github.com/nervosnetwork/ckb/tree/develop/rpc#type-indexerscriptsearchmode

The default `prefix` mode has potential exploit, malicious users may create a fake SUDT cell by appending random data after the script args.